rust/noosphere-gateway/src/worker/name_system.rs

```diff
@@ -376,3 +436,53 @@ impl<H> OnDemandNameResolver<H> {
         Ok(rx.await?)
     }
 }
+
+async fn set_counterpart_record<C, K, S>(context: C, record: &Jwt) -> Result<()>
+where
+    C: HasMutableSphereContext<K, S>,
+    K: KeyMaterial + Clone + 'static,
+    S: Storage + 'static,
+{
+    debug!("Setting counterpart record...");
+    let counterpart_identity = {
+        let sphere_context = context.sphere_context().await?;
+        let db = sphere_context.db();
+        db.require_key::<_, Did>(COUNTERPART).await?
+    };
+    let counterpart_link_record_key = format!("link_record/{counterpart_identity}");
+    let mut cursor = SphereCursor::latest(context.clone());
+    cursor
+        .write(
+            &counterpart_link_record_key,
+            &ContentType::Bytes.to_string(),
```

Comment:
Probably should make this "text/plain" (which I'm just now noticing we don't have a `ContentType` variant for :sweat: ).